pkg/operator/controller/ingress/controller.go

```diff
@@ -106,6 +106,32 @@ type reconciler struct {
 	recorder record.EventRecorder
 }
 
+// admissionRejectedError is an error type for ingresscontroller admission
+// rejections.
+type admissionRejectedError struct {
+	rejectionReason string
+}
+
+// Error returns the reason or reasons why an ingresscontroller was rejected.
+func (e *admissionRejectedError) Error() string {
+	return e.rejectionReason
+}
+
+// newAdmissionRejectedError returns a new admissionRejectedError with the given
+// rejection reason.
+func newAdmissionRejectedError(reason string) *admissionRejectedError {
+	return &admissionRejectedError{
+		rejectionReason: reason,
+	}
+}
```

Comment:
Simpler ([more idiomatic](https://golang.org/doc/effective_go.html#errors)?) to ditch the constructor:

```go
type admissionRejectedError struct {
  Reason string
}

func (e * admissionRejectedError) Error() string {
  return e.Reason
}
```

Reply:
Done.

Reply:
Should `Reason` be public?